crypto/x509/x509_v3.c

```diff
@@ -201,7 +203,7 @@ int X509_EXTENSION_set_critical(X509_EXTENSION *ex, int crit)
 {
     if (ex == NULL)
         return 0;
-    ex->critical = (crit) ? 0xFF : -1;
+    ex->critical = (crit) ? 0xFF : 0;
```

Comment:
Is `X509_EXTENSION` a private structure? If not, it seems that this is a change to the API semantics, in which case is it appropriate to backport to earlier branches?

Reply:
It is a private structure. It is declared in `x509_local.h`.

Reply:
On the other hand this can be seen as refactoring as it should have no public API impact. So I would be OK applying this to the master branch only.